pkg/cmd/dir.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"k8s.io/klog/v2"
+)
+
+// MkdirRoot creates the root configuration directory for the KCP
+// server. This has to be done early before we start bringing up server
+// components to ensure that we set the initial permissions correctly,
+// since otherwise components will create it as a side-effect.
+func MkdirRoot(dir string) error {
```

Comment:
does this have to be a public func? I only ee it used once.

Reply:
I guess not. It's public to make it a more official policy rather than just something this server does.

Reply:
With https://github.com/kcp-dev/kcp/pull/908#discussion_r853852591, can we get rid of `MkdirRoot`?